erdgo/blockchain/proxy.go

```diff
@@ -39,6 +43,30 @@ func NewElrondProxy(url string) *elrondProxy {
 	return ep
 }
 
+// ExecuteVMQuery retrieves data from existing SC trie through the use of a VM
```

Comment:
As discussed, can you also add a blocking function `WaitForNonce(address string, nonce uint64) error` and perform HTTP requests for the address nonce with a 1-second interval? The function would return `nil` if the received nonce is the same as the one from the parameter or error if timeout or other errors. The timeout could be 2 minutes

Reply:
+ `SimulateTransaction()`

Reply:
I will add **SimulateTransaction** in another PR when this route will work also this async smart contract calls